src/inc/til/u8u16convert.h

```diff
@@ -403,8 +391,8 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
     // - the resulting UTF-16 string
     // - NOTE: Throws HRESULT errors that the non-throwing sibling returns
     template<class inT>
-    typename std::enable_if<std::is_same<typename inT::value_type, char>::value, std::wstring>::type
-    u8u16(const inT in)
+    typename std::enable_if_t<std::is_same_v<typename inT::value_type, char>, std::wstring>
+    u8u16(const inT& in)

```

Comment:
I've been unable to turn the type of the parameter into a string_view. Something like ...
```
template<class charT>
u8u16(const std::basic_string_view<charT>& in)
```
... requires to actually pass a string_view. Implicit conversions e.g. from std::string wouldn't work anymore. However, at least I made the parameter a reference now.

Reply:
But why do you need this generic version in the first place? What's wrong with `std::wstring u8u16(std::string_view)` and `std::string u16u8(std::wstring_view)`?

Reply:
Are we allowed to put an not-templatized function into a header?

Reply:
Absolutely! Since this would be a wrapper function you can just mark it as `inline`, preventing conflicts. The STL does the same in `<charconv>` for instance.
For instance: https://github.com/microsoft/terminal/blob/68294f863d2adfafd6937324486b499fea73b8ab/src/inc/til/string.h#L198-L206